flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -216,4 +219,25 @@ protected void finalizeCluster() {
 	public int getMaxSlots() {
 		return 0;
 	}
+
+	private static final class RestClusterClientThreadFactory implements ThreadFactory {
```

Comment:
I think you could also use the `ExecutorThreadFactory` for this purpose.

Reply:
fixed